linker-interface/shared/src/main/scala/org/scalajs/linker/interface/ESFeatures.scala

```diff
@@ -26,36 +26,125 @@ import Fingerprint.FingerprintBuilder
  *    corresponding features if it supports them. Support for such options can
  *    be dropped in any subsequent version of the linker, including patch
  *    versions.
+ *  - Options whose name is of the form `avoidX` *hint* at the linker to avoid
+ *    the corresponding features *when it does not affect observable
+ *    semantics*. They are related to optimizations (for performance or code
+ *    size). The linker is free to ignore those options.
  */
 final class ESFeatures private (
-    /** Whether to use ECMAScript 2015 features, such as classes and arrow
-     *  functions.
-     */
-    val useECMAScript2015: Boolean,
```

Comment:
I discovered that Scaladocs on constructor parameters are ignored. Moreover, `@param` annotations in the class' Scaladoc are ignored if the constructor is not public. Therefore, I have switched to explicit `val`s in the body of the class so that they can carry proper Scaladocs that are visible online.

Reply:
> Scaladoc are ignored if the constructor is not public.

Oh, that's why. I have been toying with this as well. Seems like a Scaladoc bug to me :-/

I think this deserves a (non-scaladoc) comment in the code, otherwise it looks quite surprising IMO.